core/secrets/rotate.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package secrets
+
+// RotatePolicy defines a policy for how often
+// to rotate a secret.
+type RotatePolicy string
+
+const (
+	RotateHourly      = RotatePolicy("hourly")
+	RotateDaily       = RotatePolicy("daily")
+	RotateWeekly      = RotatePolicy("weekly")
+	RotateBiWeekly    = RotatePolicy("bi-weekly")
```

Comment:
this is ambiguous, I would prefer twice-weekly or half-monthly depending on the context intended.